Generator/Generator/TearDownInteractor.swift

```diff
@@ -0,0 +1,172 @@
+//
+//  TearDownInteractor.swift
+//  Generator
+//
+//  Created by Kryštof Matěj on 19.05.2022.
+//  Copyright © 2022 Kryštof Matěj. All rights reserved.
+//
+
+import SwiftSyntax
+import SwiftSyntaxBuilder
+import SwiftSyntaxParser
+
+public final class TearDownInteractor {
+    private let nilableVariablesParser: NilableVariablesParser
+
+    public init(nilableVariablesParser: NilableVariablesParser) {
+        self.nilableVariablesParser = nilableVariablesParser
+    }
+
+    public func execute(text: String, spacing: Int) throws -> String {
+        let file = try SyntaxParser.parse(source: text)
+        let classRewriter = ClassRewriter(
+            nilableVariablesParser: nilableVariablesParser,
+            spacing: spacing
+        )
+        return classRewriter.visit(file.root).description
+    }
+}
+
+private class ClassRewriter: SyntaxRewriter {
+    private let nilableVariablesParser: NilableVariablesParser
+    private let spacing: Int
+
+    init(nilableVariablesParser: NilableVariablesParser, spacing: Int) {
+        self.nilableVariablesParser = nilableVariablesParser
+        self.spacing = spacing
+    }
+
+    override func visit(_ node: ClassDeclSyntax) -> DeclSyntax {
+        guard isRelevantClass(node: node) else {
+            return super.visit(node)
+        }
+
+        var node = node
+
+        if let index = firstIndex(node: node, methodName: "tearDown") {
+            let newMembers = node.members.members.removing(childAt: index)
+            node.members = node.members.withMembers(newMembers)
+        }
+
+        let nilableVariables = nilableVariablesParser.parse(from: node)
+        let tearDownMethod = makeTearDownMethod(variables: nilableVariables)
+
+        if let index = firstIndex(node: node, methodName: "setUp") {
```

Comment:
Odhalí toto i funkci `func setUp(completion: @escaping (Error?) -> Void)`? Tipuju, že ano?

Reply:
Najde ji to, ale nevím zda by mělo.